src/libraries/System.Text.RegularExpressions/tests/RegexSRMTests.cs

```diff
@@ -29,6 +29,148 @@ static void WriteLine(string s)
+        private static HashSet<char>[] ComputeIgnoreCaseTable(CultureInfo culture, HashSet<char> treatedAsCaseInsensitive)
+        {
+            CultureInfo ci = CultureInfo.CurrentCulture;
+            CultureInfo.CurrentCulture = culture;
+            var ignoreCase = new HashSet<char>[0x10000];
+            for (uint i = 0; i <= 0xFFFF; i++)
+            {
+                char c = (char)i;
+                char cU = char.ToUpper(c);
+                char cL = char.ToLower(c);
+                //Turkish i without dot is only considered case-sensitive in tr and az languages
+                if (treatedAsCaseInsensitive.Contains(c) ||
+                    (c == Turkish_i_withoutDot && culture.TwoLetterISOLanguageName != "tr" && culture.TwoLetterISOLanguageName != "az"))
+                    continue;
+                if (cU != cL)
+                {
+                    var set = (ignoreCase[c] == null ? (ignoreCase[cU] == null ? (ignoreCase[cL] == null ? new HashSet<char>()
+                                                     : ignoreCase[cL]) : ignoreCase[cU]) : ignoreCase[c]);
+                    set.Add(c);
+                    set.Add(cU);
+                    set.Add(cL);
+                    ignoreCase[c] = set;
+                    ignoreCase[cL] = set;
+                    ignoreCase[cU] = set;
+                }
+            }
+            CultureInfo.CurrentCulture = ci;
+            return ignoreCase;
+        }
+
+        /// <summary>
+        /// represents the difference between the two tables as a special string
+        /// </summary>
+        private static string GetDiff(HashSet<char>[] table1, HashSet<char>[] table2)
+        {
+            List<string> diffs = new();
+            Func<HashSet<char>, int, string> F = (s, i) =>
+             {
+                 if (s == null)
+                     return ((char)i).ToString();
+                 List<char> elems = new List<char>(s);
+                 elems.Sort();
+                 string res = new string(elems.ToArray());
+                 return res;
+             };
+            for (int i = 0; i <= 0xFFFF; i++)
+            {
+                var s1 = F(table1[i], i);
+                var s2 = F(table2[i], i);
+                if (s1 != s2)
+                    diffs.Add(string.Format("{0}:{1}/{2}", (char)i, s1, s2));
+            }
+            return string.Join(",", diffs.ToArray());
+        }
+
+        /// <summary>
+        /// This test is to make sure that the generated IgnoreCaseRelation table for DFA does not need to be updated.
+        /// It would need to be updated/regenerated if this test fails.
+        /// </summary>
+        [OuterLoop("May take several seconds due to large number of cultures tested")]
+        [Fact]
+        public void TestIgnoreCaseRelation()
+        {
+            //these 22 characters are considered case-insensitive by regex, while they are case-sensitive outside regex
```

Comment:
nit, our style is a single space after //. it's not a big deal as we can bulk format them as part of cleaning up.

Reply:
I fixed some in the last pr, but more of such format changes are needed throughout the whole codebase of srm.